integration/test_auth.py

```diff
@@ -373,37 +418,41 @@ def test_import_users_with_password(api_key):
     finally:
         auth.delete_user(uid)
 
-@pytest.fixture
-def action_code_settings():
-    return auth.ActionCodeSettings('http://localhost')
-
-def _validate_link_url(link):
-    assert isinstance(link, six.string_types)
-    six.moves.urllib.parse.urlparse(link)
-
-def test_password_reset(new_user_with_params):
-    link = auth.generate_password_reset_link(new_user_with_params.email)
-    _validate_link_url(link)
-
-def test_email_verification(new_user_with_params):
-    link = auth.generate_email_verification_link(new_user_with_params.email)
-    _validate_link_url(link)
-
-def test_password_reset_with_settings(new_user_with_params, action_code_settings):
-    link = auth.generate_password_reset_link(new_user_with_params.email,
+def test_password_reset(new_user_email_unverified, api_key):
+    link = auth.generate_password_reset_link(new_user_email_unverified.email)
+    oob_code = _validate_link_url(link, check_continue_url=False)
+    assert new_user_email_unverified.email == _reset_password(oob_code, "newPassword", api_key)
```

Comment:
Lets break this down into 2 lines for clarity:

```
user_email = _reset_password(oob_code, ...)
assert new_user_email_unverified.email == user_email
```

Here and below.

Reply:
Done.